dsgrid/dataset/pivoted_table.py

```diff
@@ -0,0 +1,197 @@
+                # No work is required if the user requested the base dimension for the pivoted
+                # dimension.
+                if (
+                    query_name not in base_query_names
+                    and query_name not in context.get_dimension_query_names(pivoted_dim_type)
+                ):
+                    if column.function is not None:
+                        # TODO: Do we need to support this?
+                        raise Exception("column function cannot be set on {column}")
+                    dim = self.project_config.get_dimension(query_name)
+                    if dim.model.dimension_type == pivoted_dim_type:
+                        dimension_query_name = query_name
+                        break
+            if dimension_query_name is None:
+                continue
+            dim_config = self.project_config.get_dimension(dimension_query_name)
+            mapping_records = self.project_config.get_base_to_supplemental_mapping_records(
+                dimension_query_name
+            )
+            df, new_pivoted_columns, dropped_columns = map_and_reduce_pivoted_dimension(
+                df,
+                mapping_records,
+                pivoted_columns,
+                agg.aggregation_function.__name__,
+                rename=True,
+            )
+            context.replace_dimension_query_names(
+                dim_config.model.dimension_type, {dimension_query_name}, dataset_id=self.dataset_id
+            )
+            pivoted_columns -= dropped_columns
+            logger.info("Replaced dimensions with supplemental records %s", new_pivoted_columns)
+
+        if new_pivoted_columns:
+            context.set_pivoted_columns(new_pivoted_columns, dataset_id=self.dataset_id)
+
+        return df
+
+    def process_stacked_aggregations(
+        self, df, aggregations: List[AggregationModel], context: QueryContext
+    ):
+        """Aggregate the stacked dimensional data as specified by aggregations.
+
+        Parameters
+        ----------
+        df : pyspark.sql.DataFrame
+        aggregations : AggregationModel
+        context : QueryContext
+
+        Returns
+        -------
+        pyspark.sql.DataFrame
+
+        """
+        if not aggregations:
+            return df
+
+        group_by_query_names = []
+        pivoted_dimension_type = context.get_pivoted_dimension_type(dataset_id=self.dataset_id)
+        for agg in aggregations:
+            columns = []
+            for dim_type, column in agg.iter_dimensions_to_keep():
+                if dim_type != pivoted_dimension_type:
+                    columns.append(column)
+            if not columns:
+                continue
+            df = self.add_columns(df, columns, context, True)
+            pivoted_columns = set(context.get_pivoted_columns(dataset_id=self.dataset_id))
+            group_by_cols = []
+            for column in columns:
+                if column.dimension_query_name not in pivoted_columns:
```

Comment:
I don't think you need this because `columns` append only dims of non-pivoted dim type.